sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/ConnectionStringParser.cs

```diff
@@ -80,5 +80,25 @@ internal static string ParseTokenValue(string token)
 
             return null;
         }
+
+        /// <summary>
+        ///   It parses a fully qualified domain name and returns the contained namespace name.
+        /// </summary>
+        ///
+        /// <param name="fullyQualifiedDomainName">The FQDN in the form <c>sb://{namespacename}.servicebus.windows.net/</c></param>
+        ///
+        /// <returns>The namespace name if found; otherwise, a <see cref="InvalidOperationException" /> is thrown.</returns>
+        ///
+        internal static string ParseNamespaceName(string fullyQualifiedDomainName)
+        {
+            var regex = new Regex(@"sb://(?<namespacename>.+)\.servicebus\.windows\.net/", RegexOptions.IgnoreCase);
+
+            if (regex.IsMatch(fullyQualifiedDomainName))
+            {
+                return regex.Match(fullyQualifiedDomainName).Groups["namespacename"].Value;
+            }
+
+            throw new InvalidOperationException("The fully qualified domain name could not be parsed");
```

Comment:
Maybe `FormatException` would be better for context here?   Not that it's particularly important, given that this is test infrastructure.